.github/workflows/build.yml

```diff
@@ -299,11 +316,20 @@ jobs:
         run: |
           cat ./build/meson-logs/testlog.txt
 
+      - name: Artifact name
+        id: artifact
+        run: |
+          VERSION_FILE=$(find . -path "*/common/version.h" -type f | head -n 1)
+          VER=$(grep VERSION "$VERSION_FILE" | cut -d'"' -f2)
+          TARGET=${{ matrix.os }}${{ matrix.arch != '' && format('-{0}', matrix.arch) || '' }}
+          PR=${{ github.event.pull_request.number && format('-pr{0}', github.event.pull_request.number) || '' }}
+          echo "name=mpv-$VER$PR-${{ github.run_id }}-$TARGET" >> "$GITHUB_OUTPUT"
+
       - uses: actions/upload-artifact@v5
         if: ${{ matrix.arch != 'test' }}
         with:
-          name: mpv-${{ matrix.os }}${{ matrix.arch != '' && format('-{0}', matrix.arch) || '' }}
-          path: mpv.tar.gz
+          name: ${{ steps.artifact.outputs.name }}
+          path: build/mpv.app*
```

Comment:
at the "Create App Bundle" step you can also remove the `tar -czvf mpv.tar.gz -C build mpv.app` then.

Reply:
Done.